app/src/org/commcare/activities/UpdateActivity.java

```diff
@@ -247,6 +247,14 @@ protected void startUpdateCheck() {
         uiController.downloadingUiState();
     }
 
+    private void initUpdateTaskProgressDisplay() {
+        if (CommCareApplication._().isConsumerApp()) {
+            showProgressDialog(DIALOG_CONSUMER_APP_UPGRADE);
```

Comment:
Hm, so most of the actual dialog management, at least in terms of activity lifecycle stuff, is done by `CommCareActivity`, which `UpdateActivity` does extend. The only thing that is controlled by a `CommCareTask` is the automatic showing and dismissing of the dialog based upon the life cycle of the task, so I'm doing that part manually at the points where I know the task is starting and stopping. Agreed that it's not an amazing way to do it, but it does seem to be pretty robust (I did test it with rotating).